std/utf.d

```diff
@@ -1164,8 +1164,10 @@ package template codeUnitLimit(S)
  * Returns:
  *      decoded character
  */
-private dchar decodeImpl(bool canIndex, UseReplacementDchar useReplacementDchar = UseReplacementDchar.no, S)(auto ref S str, ref size_t index)
-    if (is(S : const char[]) || (isInputRange!S && is(Unqual!(ElementEncodingType!S) == char)))
+private RV decodeImpl(bool canIndex, UseReplacementDchar useReplacementDchar, RV = dchar, S)
```

Comment:
Except that in the case of `DecodeType.onlyValidate`, the return type is `bool`. The return type needs to be `auto` - unless we're casting to and from `dchar` for the `onlyValidate` case, which seems totally unnecessary. `decodeImpl` is a private function, and no one reading the docs is even going to see the `auto` - and plenty of functions in Phobos return `auto` anyway. So, I really don't see the problem with `decodeImpl` returning `auto`. Given that it needs to return a `dchar` in one case and a boolean value in the other, `auto` makes perfect sense.